products/BellHybrid/apps/common/src/widgets/TimeSetSpinnerVertical.cpp

```diff
@@ -23,54 +23,77 @@ namespace gui
         setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Center));
         setEdges(RectangleEdge::None);
 
-        hBoxHours = new gui::HBox(
-            this, 0, 0, style::homescreen_vertical::center_box_w, style::homescreen_vertical::center_box_h);
+        hBoxHours = new gui::HBox(this, 0, 0, style::homescreen_vertical::center_box_w, 0);
         hBoxHours->setEdges(gui::RectangleEdge::None);
         hBoxHours->setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Bottom));
+        hBoxHours->setMinimumSize(style::homescreen_vertical::center_box_w, style::homescreen_vertical::center_box_h);
 
-        hourFirst = new gui::TextFixedSize(hBoxHours, 0, 0, 0, 0);
+        hBoxHoursLeft = new gui::HBox(hBoxHours, 0, 0, 0, 0);
+        hBoxHoursLeft->setEdges(gui::RectangleEdge::None);
+        hBoxHoursLeft->setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Bottom));
+        hBoxHoursLeft->setMaximumSize(style::homescreen_vertical::side_box_w, style::homescreen_vertical::center_box_h);
+
+        hBoxHoursRight = new gui::HBox(hBoxHours, 0, 0, 0, 0);
+        hBoxHoursRight->setEdges(gui::RectangleEdge::None);
+        hBoxHoursRight->setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Bottom));
+        hBoxHoursRight->setMaximumSize(style::homescreen_vertical::side_box_w,
+                                       style::homescreen_vertical::center_box_h);
+
+        hourFirst = new gui::TextFixedSize(hBoxHoursLeft, 0, 0, 0, 0);
         updateFont(hourFirst, fontName);
 
-        hourFirst->setAlignment(Alignment(Alignment::Horizontal::Left, Alignment::Vertical::Center));
+        hourFirst->setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Center));
         hourFirst->setEdges(RectangleEdge::None);
         hourFirst->setText("0");
-        hourFirst->setMargins(getHourMargins(fontName));
+        hourFirst->drawUnderline(false);
 
-        hourSecond = new gui::TextFixedSize(hBoxHours, 0, 0, 0, 0);
+        hourSecond = new gui::TextFixedSize(hBoxHoursRight, 0, 0, 0, 0);
         updateFont(hourSecond, fontName);
 
-        hourSecond->setAlignment(Alignment(Alignment::Horizontal::Right, Alignment::Vertical::Center));
+        hourSecond->setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Center));
         hourSecond->setEdges(RectangleEdge::None);
         hourSecond->setText("0");
-        hourSecond->setMargins(getHourMargins(fontName));
+        hourSecond->drawUnderline(false);
 
-        hBoxHours->addWidget(hourFirst);
-        hBoxHours->addWidget(hourSecond);
+        hBoxHoursLeft->addWidget(hourFirst);
+        hBoxHoursRight->addWidget(hourSecond);
```

Comment:
is adding those widgets neccessary? `hourFirst` and `hourSecond` are constructed with parents `hBoxHoursLeft` and `hBoxHoursRight` already. Maybe we can omit those?

Reply:
Removed. Good point